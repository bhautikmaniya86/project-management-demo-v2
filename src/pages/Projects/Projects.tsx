import React from 'react';
import ProjectList from './ProjectList/ProjectList';

const Projects = () => {
  return <ProjectList />;
};

export default Projects;
