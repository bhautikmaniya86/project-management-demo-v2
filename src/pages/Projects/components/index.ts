export { default as ProjectForm } from './ProjectForm/ProjectForm';
