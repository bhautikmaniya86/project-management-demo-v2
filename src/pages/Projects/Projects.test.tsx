import { render, screen } from '@testing-library/react';
import React from 'react';
import Projects from './Projects';

// Mock the ProjectList component
jest.mock('./ProjectList/ProjectList', () => {
  return function MockProjectList() {
    return <div data-testid="mock-project-list">Mock Project List</div>;
  };
});

describe('Projects Component', () => {
  it('renders without crashing', () => {
    render(<Projects />);
    expect(screen.getByTestId('mock-project-list')).toBeInTheDocument();
  });

  it('renders the ProjectList component', () => {
    render(<Projects />);
    expect(screen.getByText('Mock Project List')).toBeInTheDocument();
  });

  it('does not render any additional content', () => {
    const { container } = render(<Projects />);
    expect(container.firstChild).toMatchSnapshot();
  });

});
