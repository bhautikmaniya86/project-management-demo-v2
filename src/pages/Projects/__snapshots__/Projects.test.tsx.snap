// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Component does not render any additional content 1`] = `
<div
  data-testid="mock-project-list"
>
  Mock Project List
</div>
`;
